distribution/tools/entitlements-agent/build.gradle

```diff
@@ -0,0 +1,48 @@
+import static java.util.stream.Collectors.joining
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+apply plugin: 'elasticsearch.build'
+
+base {
+  archivesName = 'elasticsearch-entitlements-agent'
+}
+
+configurations {
+  entitlementRuntime
+}
+
+dependencies {
+  entitlementRuntime project(":libs:elasticsearch-entitlements-runtime")
+  implementation project(":libs:elasticsearch-entitlements-runtime")
+  implementation 'org.ow2.asm:asm:9.7'
+  testImplementation project(":test:framework")
+}
+
+tasks.named("test").configure {
+  systemProperty "tests.security.manager", "false"
+}
+
+test {
+  dependsOn(jar)
+  def runtimeJars = configurations.entitlementRuntime.files.stream().map(File::toString).collect(joining(File.pathSeparator))
```

Comment:
What's the long-term plan here? We certainly don't want to have to pass this system property into ES in production. If we just need the path to the jar file couldn't we look it up via reflection (i.e. `getCodeSource()`)?

Reply:
I'm not actually sure. I used this because I couldn't get anything else working, but I'm definitely open to suggestions.

Reply:
Assuming the runtime jar will be on the classpath, which I presume it will need to be if any other code is to interface with it, then you should just be able to lookup any class in that jar and find it's codesource to get the path to the jar. I'm certain we do similar things elsewhere.

Reply:
Thanks Mark, that worked.